packages/eas-cli/src/build/utils/repository.ts

```diff
@@ -83,6 +83,36 @@ export async function commitPromptAsync(
   });
 }
 
+export async function makeProjectMetadataFileAsync(
+  archivePath: string
+): Promise<{ path: string; size: number }> {
+  const spinner = ora('Creating project metadata file');
+  const timerLabel = 'makeProjectMetadataFileAsync';
+  startTimer(timerLabel);
+
+  const metadataLocation = path.join(getTmpDirectory(), `${uuidv4()}.json`);
+  const archiveContent: string[] = [];
+
+  await tar.list({
+    file: archivePath,
+    onentry: (entry: tar.ReadEntry) => {
+      if (entry.type === 'File' && !entry.path.includes('.git')) {
+        archiveContent.push(entry.path);
+      }
+    },
+  });
+
+  await fs.writeJSON(metadataLocation, {
+    archiveContent,
+  });
```

Comment:
maybe we want to wrap it in try-catch and add logic similar to https://github.com/expo/eas-cli/pull/2149/files#diff-877970adc4c19b7884f2246d873bbfd7a36bd66a803ee839566585499c45a77fR144-R149